src/layouts/ResourceRoom/ResourceRoom.tsx

```diff
@@ -80,40 +91,73 @@ const EmptyResourceRoom = () => {
+      <Box as="form" w="full" mt={100}>
         {/* Resource Room does not exist */}
-        <Section align="flex-start">
-          <SectionHeader label="Create Resource Room" />
-          <SectionCaption label="NOTE: " icon={BiInfoCircle}>
-            You must create a Resource Room before you can create Resources.
-          </SectionCaption>
-          {/* Info segment */}
-          <FormControl
-            isRequired
-            isInvalid={!!errors.newDirectoryName?.message}
-          >
-            <Input
-              maxW="50%"
-              {...register("newDirectoryName", { required: true })}
-            />
-            <FormErrorMessage>
-              {errors.newDirectoryName?.message}
-            </FormErrorMessage>
-          </FormControl>
-          {/* Segment divider  */}
-          <Divider color="border.divider.alt" />
+        <VStack spacing={5}>
+          <EmptyBoxImage />
+          <VStack spacing={0}>
+            <Center textStyle="subhead-1">
+              There&apos;s nothing here yet.
+            </Center>
+            <Center textStyle="body-2">
+              Create a resource room to get started.
+            </Center>
+          </VStack>
           <Button
-            type="submit"
-            isLoading={isLoading}
-            onClick={handleSubmit(onSubmit)}
-            isDisabled={!_.isEmpty(errors)}
+            variant="solid"
+            onClick={onOpen}
+            leftIcon={<Icon as={BiPlus} fontSize="1.5rem" fill="white" />}
           >
             Create Resource Room
           </Button>
-        </Section>
+        </VStack>
       </Box>
+
+      <Modal blockScrollOnMount={false} isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Create resource room</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Text textStyle="subhead-1">Resource room title</Text>
+
+            <FormControl
+              isRequired
+              isInvalid={!!errors.newDirectoryName?.message}
+            >
+              <Input
+                marginTop={5}
+                placeholder="Resource room name"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register("newDirectoryName", {
+                  required: "Please enter resource room name",
+                })}
+              />
+              <FormErrorMessage>
+                {errors.newDirectoryName?.message}
+              </FormErrorMessage>
+            </FormControl>
+          </ModalBody>
+          <ModalFooter>
+            <Button mr={3} variant="link" onClick={onClose}>
+              <Text textStyle="subhead-1" colorScheme="orange">
+                Cancel
+              </Text>
+            </Button>
+            <LoadingButton
+              isDisabled={!_.isEmpty(errors)}
+              type="submit"
+              onClick={handleSubmit(onSubmit)}
+            >
```

Comment:
oops, could we use `Button` and pass the `isLoading` prop? this component is legacy (should've marked it in JSDoc as dep also psps) and it's not ideal to use it.

Reply:
could we do this^

Reply:
Actually this is because the component has crashed before due to the `useEffect` ._.